prometheus/counter.go

```diff
@@ -136,7 +137,8 @@ func NewCounterVec(opts CounterOpts, labelNames []string) *CounterVec {
 	return &CounterVec{
 		metricVec: newMetricVec(desc, func(lvs ...string) Metric {
 			if len(lvs) != len(desc.variableLabels) {
-				panic(errInconsistentCardinality)
+				err := fmt.Errorf("%s: labels and values for '%s' don't match: labels %v, values %v", errInconsistentCardinality, desc.fqName, desc.variableLabels, lvs)
```

Comment:
This will expand to something like
```
inconsistent label cardinality: labels and values for 'requests_total' don't match: labels [foo bar], values [baz]
```

which is not as precise as it could be and has a bit of stutter with two colons and such.

How about the following:

```Go
panic(fmt.Errorf(
    "%s: %q has %d variable labels named %q but %d values %q were provided",
    errInconsistentCardinality, desc.fqName,
    len(desc.variableLabels), desc.variableLabels,
    len(lvs), lvs,
))
```

That would result in:
```
inconsistent label cardinality: "requests_total" has 2 variable labels named ["foo" "bar"] but only 1 values ["baz"] were provided
```

Besides the usual singular/plural oddness, I think this tells you more precisely what the actual issue is and has nicer quoting.

Reply:
Much better -thanks!